lib/roast/workflow/workflow_executor.rb

```diff
@@ -204,12 +262,88 @@ def strip_and_execute(step)
 
           # NOTE: We don't need to call interpolate here as it's already been done
           # in execute_string_step before this method is called
-          %x(#{command})
+          begin
+            output = %x(#{command})
+            raise CommandExecutionError.new("Command exited with non-zero status", step_name: command) unless $CHILD_STATUS.success?
```

Comment:
This should be configurable on the step. I have some commands that I execute so I can get the failure output as context for what the workflow should do next (example: run a failing typecheck to get the output of "these are the problems to fix"). 

In those cases, I don't care if the command has a successful exit code and want to continue regardless.